tests/test_providers.py

```diff
@@ -711,7 +709,11 @@ def test_is_base_available(
     base = Base(name=name, channel=channel, architectures=architectures)
     valid, reason = providers.is_base_available(base)
 
-    assert (valid, reason) == (expected_valid, expected_reason)
+    assert valid == expected_valid
+    if reason is None:
+        assert expected_reason == reason
+    else:
+        assert re.match(expected_reason, reason)
```

Comment:
```suggestion
        assert re.fullmatch(expected_reason, reason)
```

This will ensure that the whole string matches the regex rather than a substring